### PR TITLE
Fixed test and validation sets for moons dataset

### LINKED ISSUE #6: Fix the Test and Validation set sizes
Currently, Test and Validation set sizes are also randomly generated. For comparing results across runs both the test and validation set sizes should be static. 

- Test set size should be 1000
- Validation set size should also be 1000

Sklearn datasets that should have static test and validation set sizes are:
- moons dataset
- circles dataset
- make blobs

### PR DESCRIPTION
**Pull request by RafayAK**
- create fixed validation and test sets across runs. Both validation and test sets have 1000 examples, each
- change parameter distribution plots from KED only to Histogram + KDE, for better visualization
- clear plots to free up memory 
- save parameter values as NumPy files for plotting later
- fix the f1 score according to binary classification
- update PyTorch to use Cuda 11.6
- update bash script for moons experiment.

### CODE REVIEW

**Comment by RafayAK** [comment]
Wandb dashboard as a result of `moons_run.sh`
https://wandb.ai/rafay/moons_experiment_3?workspace=user-rafay

**Comment by RafayAK** [comment]
merging changes, as most are just some code refactor or update

**Review by alexhernandezgarcia — COMMENTED** [review]
Sorry for the late review It all looks good to me!

File: configs/datamodule/moons.yaml [hunk 1]
```diff
@@ -1,8 +1,20 @@
 _target_: src.datamodules.sklearn_datamodule.create_sklearn_datamodule
-dataset:
+
+train_dataset:
```
Comment (JARVVVIS) [inline configs/datamodule/moons.yaml:3]: Since Train, Val, and Test datasets are initialized separately using `sklearn.datasets.make_moons`, there could be data leakage across the splits. I am working on a potential fix for this, perhaps we can discuss more in the meeting.
